src/components/CheckBox/CheckBox.style.ts

```diff
@@ -89,18 +87,27 @@ export const checkboxStyle = ({ intermediate, checked, filled }: Props) => (
     &:disabled + label {
       cursor: not-allowed;
     }
+  `;
+};
 
-    // Checkmark.
-    &:checked + label:after {
-      content: '';
+export const markerStyle = ({ intermediate, checked, filled }: Props) => (
+  theme: Theme
+): SerializedStyles => {
+  const symbolColor = getSymbolColor({ intermediate, checked, filled, theme });
+
+  return css`
+    span {
+      padding: 0;
+    }
+
+    svg {
       position: absolute;
-      left: ${rem(5.5)};
-      ${intermediate
-        ? `width: ${rem(13)}; top: ${rem(10.5)}; height: ${rem(3)};`
-        : `width: ${rem(2)}; top: ${rem(12)}; height: ${rem(2)};;
-      box-shadow: -2px 0 0 ${symbolColor}, 2px 0 0 ${symbolColor}, 4px 0 0 ${symbolColor}, 4px -2px 0 ${symbolColor}, 4px -4px 0 ${symbolColor}, 4px -6px 0 ${symbolColor}, 4px -8px 0 ${symbolColor}, 4px -10px 0 ${symbolColor};
-      transform: rotate(45deg);`};
-      background-color: ${symbolColor};
+      top: 0;
+      display: ${checked ? 'block' : 'none'};
+    }
+
+    path {
+      fill: ${symbolColor} !important;
```

Comment:
To avoid using the `!important` keyword you can use the getSymbolColor directly in Icon's `color` prop.

example:
```
const theme = useTheme();
...

    <Icon
            name={intermediate ? 'minus' : 'checkmark'}
            size={24}
            color={getSymbolColor({ intermediate, checked, filled, theme })}
          />
```